src/websocket_endpoint.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "websocketpp/client.hpp"
+#include "websocketpp/config/asio_no_tls_client.hpp"
+#include "websocketpp/extensions/permessage_deflate/enabled.hpp"
+#include <functional>
+#include <thread>
+#include <memory>
+namespace sensr
+{
+    typedef websocketpp::client<websocketpp::config::asio_client> websocketpp_client;
+    class WebSocketEndPoint
+    {
+    public:
+        using MsgReceiver = std::function<void(const std::string& msg)>;
+        using ErrorReceiver = std::function<void(const std::string& err)>;
+        WebSocketEndPoint();
+        ~WebSocketEndPoint();
+
+        bool Connect(const std::string &uri, MsgReceiver func, ErrorReceiver err_func);
+        void Close(websocketpp::close::status::value code);
+
+    private:
+        void OnOpen(websocketpp_client *c, websocketpp::connection_hdl hdl);
+        void OnFail(websocketpp_client *c, websocketpp::connection_hdl hdl);
+        void OnClose(websocketpp_client *c, websocketpp::connection_hdl hdl);
+        void OnMessage(websocketpp::connection_hdl hdl, websocketpp_client::message_ptr msg);
+        enum struct Status {
+            kConnecting,
+            kOpen,
+            kClosed,
+            kFailed
+        };
+        websocketpp_client endpoint_;
+        std::shared_ptr<std::thread> thread_;
```

Comment:
Is this going to be shared with other objects? If not, let's use `std::unique_ptr`.

Reply:
Change to normal member variable. Because it doesn't need to be pointer.